<!--
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <include file="$(find turtlebot_navigation)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find turtlebot_navigation)/launch/includes/safety_controller.launch.xml"/>

  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="map_topic" default="map" />
  <arg name="custom_param_file" default="$(find target_scan_follower)/param/costmap_common_params.yaml"/>

  <!-- This param is for TURTLEBOT -->
  <node pkg="move_base_correction" type="move_base" respawn="false" name="move_base_correction" ><!--output="screen"-->
    <rosparam file="$(find target_scan_follower)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find target_scan_follower)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find target_scan_follower)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find target_scan_follower)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find target_scan_follower)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find target_scan_follower)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find target_scan_follower)/param/global_planner_params.yaml" command="load" /><!-- global_planner_params ? -->
    <rosparam file="$(find target_scan_follower)/param/navfn_global_planner_params.yaml" command="load" /><!-- global_planner_params ? --><!-- params are too mauch ? -->
    <!-- external params file that could be loaded into the move_base namespace -->
    <rosparam file="$(arg custom_param_file)" command="load" />

    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
    <remap from="map" to="$(arg map_topic)" />
  </node>
</launch>
